internal/socket/utils.go

```diff
@@ -54,3 +56,10 @@ func GenerateToken(len int) (string, error) {
 	withEqualses := base64.URLEncoding.EncodeToString(b)
 	return strings.TrimRight(withEqualses, "="), nil // Trim the equals signs because they're not valid in env vars
 }
+
+// WriteError writes an error as an ErrorResponse (JSON-encoded). The err value
+// is converted to a string with fmt.Sprint.
+func WriteError(w http.ResponseWriter, err any, code int) {
+	w.WriteHeader(code)
+	json.NewEncoder(w).Encode(ErrorResponse{Error: fmt.Sprint(err)})
```

Comment:
should we check the error returned by encode? i think it's probably not worth bothering, but perhaps worth thinking about?

if not, i'm a proponent of
```suggestion
	_ = json.NewEncoder(w).Encode(ErrorResponse{Error: fmt.Sprint(err)})
```

Reply:
So, I did it (and added logging for a couple of other errors). It's... fine.